internal/mempool/mempool.go

```diff
@@ -888,10 +891,13 @@ func (txmp *TxMempool) purgeExpiredTxs(blockHeight int64) {
 	}
 
 	for _, wtx := range expiredTxs {
-		txmp.removeTx(wtx, false)
+		txmp.removeTx(wtx, true)
```

Comment:
why are you now removing it from the cache? Why was it false before?

Reply:
If it's not removed from cache, then an expired transaction will get the already-in-cache error if it is re-attemped. I ran into this when trying to backfill nonces that had expired.

Reply:
(I suspect it was originally an oversight but we should get @codchen's confirmation)

Reply:
we can change it to `!txmp.config.KeepInvalidTxsInCache` similar to how `removeTx` is called in `handleRecheckResult`. See my comment on `cache` above but we'd like to allow node operator to control whether they want to give invalid tx (whether rejected by checktx, expired, or something else) a "second chance"